### PR TITLE
created fe update profile with integration

### PR DESCRIPTION
**Pull request by akhilarjun3-sureify**
Screenshot from 2023-06-20 11-45-48
Screenshot from 2023-06-20 11-45-57
Screenshot from 2023-06-20 11-46-36
Screenshot from 2023-06-20 11-46-53
Screenshot from 2023-06-20 11-47-55

### CODE REVIEW

File: src/components/routes/AppRoutes.jsx [hunk 1]
```diff
@@ -67,6 +68,7 @@ const AppRoutes = () => {
                                 path="answeredpolls/:id"
                                 element={<ViewSinglePoll />}
                             />
+                            <Route path="profile" element={<UpdateProfile />}/>
```
Comment (harshavardhan-sureify) [inline src/components/routes/AppRoutes.jsx:71]: add profile to admin too

File: src/components/user/UpdateProfile.jsx [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+  };
+  const fetchData = async () => {
+    try {
+      const res = await axios.get(userInfo, {
+        headers: {
+          Authorization: `Bearer ${user.token}`,
+        },
+      });
+      if (res.status === 200) {
+        setuserData({ ...res.data.data.user, password: "3210@Abcd" });
```
Comment (maniknr-sureify) [inline src/components/user/UpdateProfile.jsx:177]: what is password here ? hardcoded value?

Comment (akhilarjun3-sureify) [reply src/components/user/UpdateProfile.jsx:177]: Yes.It is not appropriate to fetch password.So It is like a dummy password which will be hidden in the fe.

File: src/components/routes/AppRoutes.jsx (hunk 1 above)
Comment (akhilarjun3-sureify) [reply src/components/routes/AppRoutes.jsx]: resolved.

**Review by maniknr-sureify — APPROVED** [review]

**Review by harshavardhan-sureify — APPROVED** [review]

File: src/components/routes/AppRoutes.jsx [hunk 3]
```diff
@@ -9,71 +9,55 @@ import { ActivePolls } from "../user/ActivePolls";
...
+        <Route element={<PublicRoute />}>
+          <Route path="/login" element={<LoginPage />} />
+          <Route path="/signup" element={<SignupPage />} />
+        </Route>
+        <Route element={<ProtectedRoute />}>
+          {user.role === ADMIN_ROLE && (
+            <Route path="/admin" element={<LandingPage />}>
+              <Route path="dashboard" element={<AdminDashBoard />} />
+              <Route path="create" element={<PollCreate />} />
+              <Route path="trigger" element={<Trigger />} />
```
Comment (abhishekdundigalla-sureify) [inline src/components/routes/AppRoutes.jsx:38]: which functionality of the application is part of the Trigger component ?

File: src/components/routes/AppRoutes.jsx [hunk 4]
```diff
@@ -9,71 +9,55 @@ import { ActivePolls } from "../user/ActivePolls";
...
+            </Route>
+          )}
+          {user.role === USER_ROLE && (
+            <Route path="/user" element={<LandingPage />}>
+              {/* <Route index element={<PageNotFound />} /> */}
+              <Route path="dashboard" element={<ActivePolls />} />
+              <Route path="dashboard/:id" element={<ViewSinglePoll />} />
+              <Route path="answeredpolls" element={<AnsweredPolls />} />
+              <Route path="answeredpolls/:id" element={<ViewSinglePoll />} />
+              <Route path="profile" element={<UpdateProfile />} />
```
Comment (abhishekdundigalla-sureify) [inline src/components/routes/AppRoutes.jsx:53]: Rename UpdateProfile component to Profile

File: src/components/routes/AppRoutes.jsx [hunk 5]
```diff
@@ -9,71 +9,55 @@ import { ActivePolls } from "../user/ActivePolls";
...
 import ProtectedRoute from "./ProtectedRoute";
 import { auth } from "../features/User.reducer";
 import { useSelector } from "react-redux";
 import PublicRoute from "./PublicRoute";
-import AllUsers from "./../admin/AllUsers"
+import AllUsers from "./../admin/AllUsers";
 import AllPolls from "../admin/AllPolls";
-import {AdminActivePolls} from "./../admin/AdminActivePolls"
+import { AdminActivePolls } from "./../admin/AdminActivePolls";
+import UpdateProfile from "../user/UpdateProfile";
```
Comment (abhishekdundigalla-sureify) [inline src/components/routes/AppRoutes.jsx:21]: rename UpdateProfile component name to Profile

File: src/components/user/UpdateProfile.jsx [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+          message =
+            "Password must contain atleast one upper,one lower,one digit and one special character";
+        }
+        break;
+      default:
+    }
+
+    setErrors({ ...errors, [clickedField]: message });
+  };
+  const updateUser = async (data) => {
```
Comment (abhishekdundigalla-sureify) [inline src/components/user/UpdateProfile.jsx:92]: change updateUser function name to a meaningful one

File: src/components/user/UpdateProfile.jsx [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+        headers: { Authorization: `Bearer ${user.token}` },
+      });
+      if (res.status === 200) {
+        setEnabled(true);
+        handleClose();
+        setPassword("");
+      }
+    } catch (error) {
+      if (error.response.data.status === 400) {
+        setToaster("Wrong password");
```
Comment (abhishekdundigalla-sureify) [inline src/components/user/UpdateProfile.jsx:160]: do not use toasters for invalid / incorrect form fields, use them only for the actions like profile updated successfully / technical error.
you can directly display the form error messages in the form itself on top of above all form fields.

File: src/components/user/UpdateProfile.jsx [hunk 8]
```diff
@@ -0,0 +1,332 @@
...
+      } else {
+        setToaster("Something went Wrong");
```
Comment (abhishekdundigalla-sureify) [inline src/components/user/UpdateProfile.jsx:162]: i think else case is not required, check once

File: src/components/user/UpdateProfile.jsx (hunk 2 above)
Comment (abhishekdundigalla-sureify) [inline src/components/user/UpdateProfile.jsx:177]: do not use hard-coded password value

File: src/components/user/UpdateProfile.jsx [hunk 9]
```diff
@@ -0,0 +1,332 @@
...
+      });
+      if (res.status === 200) {
+        setuserData({ ...res.data.data.user, password: "3210@Abcd" });
+      }
+    } catch (error) {
+      setToaster(error.response.data.error);
+    }
+  };
+  useEffect(() => {
+    fetchData();
```
Comment (abhishekdundigalla-sureify) [inline src/components/user/UpdateProfile.jsx:184]: change fetchData function name to a meaningful one

File: src/constants.js [hunk 10]
```diff
@@ -7,57 +7,63 @@ const addUser = PHP_BASE_URL + "adduser";
 const ACTIVE_POLLS_URL = PHP_BASE_URL + "activepolls";
 const ANSWERED_POLLS_URL = PHP_BASE_URL + "answeredpolls";
 const loginApi = GO_BASE_URL + "login";
-const adminDashboardApi = GO_BASE_URL+"admin/dashboard";
+const adminDashboardApi = GO_BASE_URL + "admin/dashboard";
 const allUsers = PHP_BASE_URL + "viewusers";
+const userInfo = GO_BASE_URL + "user/userinfo";
 export const SAVE_POLL_URL = PHP_BASE_URL + "savepoll";
+const confirmuser = GO_BASE_URL + "user/confirmuser";
+const updateProfile = GO_BASE_URL + "user/updateprofile";
```
Comment (abhishekdundigalla-sureify) [inline src/constants.js:15]: rename updateProfile to updateUserProfile

File: src/constants.js [hunk 11]
```diff
@@ -7,57 +7,63 @@ const addUser = PHP_BASE_URL + "adduser";
...
 export const ADMIN_ROUTES = [
```
Comment (abhishekdundigalla-sureify) [inline src/constants.js:16]: why and where ADMIN_ROUTES are being used ?

**Review by abhishekdundigalla-sureify — CHANGES_REQUESTED** [review]
Complete the required changes

File: src/components/routes/AppRoutes.jsx (hunk 3 above)
Comment (akhilarjun3-sureify) [reply src/components/routes/AppRoutes.jsx:38]: changed

File: src/components/routes/AppRoutes.jsx [hunk 12]
```diff
@@ -9,71 +9,55 @@ import { ActivePolls } from "../user/ActivePolls";
...
+              <Route path="allusers" element={<AllUsers />} />
+              <Route path="allpolls" element={<AllPolls />} />
+              <Route path="activepolls" element={<AdminActivePolls />} />
+              <Route path="activepolls/:id" element={<ViewSinglePoll />} />
+              <Route path="profile" element={<UpdateProfile />} />
+            </Route>
+          )}
+          {user.role === USER_ROLE && (
+            <Route path="/user" element={<LandingPage />}>
+              {/* <Route index element={<PageNotFound />} /> */}
```
[the remark this answers is not in the document]
Comment (akhilarjun3-sureify) [reply src/components/routes/AppRoutes.jsx]: Removed

File: src/components/routes/AppRoutes.jsx (hunk 5 above)
Comment (akhilarjun3-sureify) [reply src/components/routes/AppRoutes.jsx]: renamed

File: src/components/user/UpdateProfile.jsx (hunk 6 above)
Comment (akhilarjun3-sureify) [reply src/components/user/UpdateProfile.jsx]: changed

File: src/components/user/UpdateProfile.jsx (hunk 7 above)
Comment (akhilarjun3-sureify) [reply src/components/user/UpdateProfile.jsx]: changed

File: src/components/user/UpdateProfile.jsx (hunk 8 above)
Comment (akhilarjun3-sureify) [reply src/components/user/UpdateProfile.jsx]: changed

File: src/components/user/UpdateProfile.jsx (hunk 2 above)
Comment (akhilarjun3-sureify) [reply src/components/user/UpdateProfile.jsx]: changed

File: src/constants.js (hunk 10 above)
Comment (akhilarjun3-sureify) [reply src/constants.js]: changed

File: src/components/user/UpdateProfile.jsx [hunk 13]
```diff
@@ -0,0 +1,332 @@
...
+      const res = await axios.get(userInfo, {
+        headers: {
+          Authorization: `Bearer ${user.token}`,
+        },
+      });
+      if (res.status === 200) {
+        setuserData({ ...res.data.data.user, password: "3210@Abcd" });
+      }
+    } catch (error) {
+      setToaster(error.response.data.error);
```
[the remark this answers is not in the document]
Comment (akhilarjun3-sureify) [reply src/components/user/UpdateProfile.jsx]: changed

**Review by abhishekdundigalla-sureify — APPROVED** [review]

**Review by Madhukar-Sureify — APPROVED** [review]

**Pull request merged**